lightning/src/ln/channel.rs

```diff
@@ -2363,12 +2405,17 @@ impl<Signer: Sign> Channel<Signer> {
 				}
 				bitcoin_tx.txid
 			};
-			let htlcs_cloned: Vec<_> = commitment_tx.2.iter().map(|htlc| (htlc.0.clone(), htlc.1.map(|h| h.clone()))).collect();
-			(commitment_tx.1, htlcs_cloned, commitment_tx.0, commitment_txid)
+			let htlcs_cloned: Vec<_> = commitment_tx.3.iter().map(|htlc| (htlc.0.clone(), htlc.1.map(|h| h.clone()))).collect();
+			(commitment_tx.2, htlcs_cloned, commitment_tx.0, commitment_txid, commitment_tx.1)
 		};
 
+		// If our counterparty updated the channel fee in this commitment transaction, check that
+		// they can actually afford the new fee now.
+		let update_fee = if let Some((_, update_state)) = self.pending_update_fee {
+			update_state == FeeUpdateState::RemoteAnnounced
+		} else { false };
+		if update_fee { debug_assert!(!self.is_outbound()); }
```

Comment:
Do we have a corresponding check that the local can afford X new feerate?

Reply:
Grrr, wow we don't test for that at all. Given we're still strongly recommending testnet for 0.0.100, I think we should get it out the door and deal with this in 0.0.101, which can hopefully be a quick ship. #1041